src/ImageSharp/Formats/Bmp/BmpEncoderCore.cs

```diff
@@ -54,96 +50,58 @@ public void Encode<TPixel>(Image<TPixel> image, Stream stream)
             int bytesPerLine = 4 * (((image.Width * bpp) + 31) / 32);
             this.padding = bytesPerLine - (image.Width * (int)this.bitsPerPixel);
 
-            // Do not use IDisposable pattern here as we want to preserve the stream.
-            var writer = new EndianBinaryWriter(Endianness.LittleEndian, stream);
-
-            var infoHeader = new BmpInfoHeader
-            {
-                HeaderSize = BmpInfoHeader.BitmapInfoHeaderSize,
-                Height = image.Height,
-                Width = image.Width,
-                BitsPerPixel = bpp,
-                Planes = 1,
-                ImageSize = image.Height * bytesPerLine,
-                ClrUsed = 0,
-                ClrImportant = 0
-            };
+            var infoHeader = new BmpInfoHeader(
+                headerSize: BmpInfoHeader.Size,
+                height: image.Height,
+                width: image.Width,
+                bitsPerPixel: bpp,
+                planes: 1,
+                imageSize: image.Height * bytesPerLine,
+                clrUsed: 0,
+                clrImportant: 0);
 
             var fileHeader = new BmpFileHeader(
                 type: 19778, // BM
                 offset: 54,
                 reserved: 0,
                 fileSize: 54 + infoHeader.ImageSize);
 
-            WriteHeader(writer, fileHeader);
-            this.WriteInfo(writer, infoHeader);
-            this.WriteImage(writer, image.Frames.RootFrame);
+            byte[] buffer = new byte[40]; // TODO: stackalloc
```

Comment:
It's probably better to actually do the stackalloc now. We'll forget otherwise.

Reply:
Annoyingly, we can't use stack allocated memory (prior to NETCORE 2.1) for the stream reads. Removing these comments for now.